src/ApiClient.php

```diff
@@ -39,17 +40,32 @@ class ApiClient
      */
     private $userAgent;
 
+    private CookieJar $cookieJar;
+    private $cookiesCacheFile = __DIR__.'/cookies.txt';
+    private $crumbCacheFile = __DIR__.'/crumb.txt';
+
     public function __construct(ClientInterface $guzzleClient, ResultDecoder $resultDecoder)
     {
         $this->client = $guzzleClient;
         $this->resultDecoder = $resultDecoder;
         $this->userAgent = UserAgent::getRandomUserAgent();
+        $this->client = new Client([
+            'headers' => [
+                'User-Agent' => $this->userAgent,
+                'Accept' => 'application/json, text/plain, */*',
+                'Accept-Language' => 'en-US,en;q=0.9',
+                'Referer' => 'https://finance.yahoo.com/',
+                'Origin' => 'https://finance.yahoo.com',
+            ],
+            'http_errors' => false,
+        ]);
     }
 
     public function getHeaders(): array
     {
         return [
-            'User-Agent' => $this->userAgent,
+            'User-Agent' => 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36',
+            // 'Accept' => 'text/html,application/json;q=0.9;*/*',
```

Comment:
So you're hardcoding a Chrome 114 user agent.
- Why Chrome 114? (which is almost 2 years old)
- Why here? You could also do it in the constructor, when setting all the default header values.